store/src/java-test/com/zimbra/cs/mailbox/DumpsterTest.java

```diff
@@ -115,9 +115,7 @@ public void moveDumpsterDoc() throws Exception {
             if (!immutableException) {
                 throw e;
             }
-        } finally {
-            mbox.endTransaction(success);
```

Comment:
You removed a call to `endTransaction`? But there is no corresponding transaction commit/close in the new code? Please explain why this is the correct change to make here.

Reply:
Added an assertion to test whether or not there is an active transaction at this point, a quick code review makes me think that there should not be.

Reply:
I removed mbox.endTransaction(success) because new implementarion calls close implicitly, but you're right, I forgot to call commit method.